File: python/leetcode/LEETCODE_83.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        if not head:
            return head
        
        prev: Optional[ListNode] = head        
        p: Optional[ListNode] = prev.next
        
        while p:
            
            if prev.val == p.val:
                prev.next = p.next
            else:
                prev = prev.next
            
            p = prev.next
            
        return head

File: java/leetcode/LEETCODE_83.java
package leetcode;


import java.util.Objects;

class LEETCODE_83 {
    public ListNode deleteDuplicates(ListNode head) {
        
        if (Objects.isNull(head)) {
            return head;
        }
        
        ListNode prev = head;
        ListNode p = prev.next;
        
        while (!Objects.isNull(p)) {
            
            if (prev.val == p.val) {
                prev.next = p.next;
            } else {
                prev = prev.next;                
            }
            
            p = prev.next;
        }
            
        return head;
    }
}